### ISSUE TITLE
bug: ios 14, toggling pointer events does not re-enable scrolling

### ISSUE THREAD

**Issue by sylwester-** [ionic-team/ionic-framework#22306]
**Ionic version:**
[x] **5.x**

**Current behavior:**
On iOS device, the side menu is opened and ion-input is tapped on (does not have to be changed), after closing the side menu, scroll does not work on current page. 

**Expected behavior:**
Scroll should work normally

**Steps to reproduce:**

1. Navigate to `https://github.com/ionic-team/ionic-conference-app` and install app as per documentation attached. (Optionally `npm update` can be performed.)
2. Build and deploy to iOS device
3. Open "About" page [important]
4. Open ion-menu in the right top corner
5. Enable "Dark Mode" (any other ion-input also will work)
6. Close ion-menu
7. Without tapping on any ion-input on the "About" page try to scroll.

This issue can be reproduced easily on any page with content, provided that user does not touch an actionable component while trying to scroll. A simple page with static ion-cards will also show this issue.

**Related code:**
[https://github.com/ionic-team/ionic-conference-app](https://github.com/ionic-team/ionic-conference-app)

A sample application via GitHub:

[https://github.com/ionic-team/ionic-conference-app](https://github.com/ionic-team/ionic-conference-app)

**Confirmed devices & iOS versions**
1. iPhone 11 Pro - iOS 14.0.1
2. iPhone 6 - iOS 12.4.8
3. iPad Pro (11-inch) - iPadOS 14.0.1

```
Ionic:

   Ionic CLI                     : 6.11.11 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.1.1
   @angular-devkit/build-angular : 0.803.25
   @angular-devkit/schematics    : 8.3.25
   @angular/cli                  : 8.3.25
   @ionic/angular-toolkit        : 2.2.0

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : ios 5.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.1.3, (and 5 other plugins)

Utility:

   cordova-res (update available: 0.15.1) : 0.14.0
   native-run                             : not installed

System:

   Android SDK Tools : 26.1.1 (/Users/sylwester/Library/Android/sdk)
   ios-deploy        : 1.10.0
   ios-sim           : 8.0.2
   NodeJS            : v12.18.4 (/usr/local/Cellar/node@12/12.18.4/bin/node)
   npm               : 6.14.6
   OS                : macOS Big Sur
```
![GIF image-FBCC53CA96AA-1](https://user-images.githubusercontent.com/8246685/95871948-8db5d600-0d6e-11eb-8199-89ced3f7758d.gif)

**Comment by liamdebeasi** [comment]
Thanks for the issue. What version of iOS are you testing this on? I am not able to reproduce this on my devices.

**Comment by sylwester-** [comment]
@liamdebeasi - it's 14.0.1. Latest official release.

I also have another device with 14.2 beta and this issue is also there.

**Comment by liamdebeasi** [comment]
Ok thanks. Which device are you testing this on? I tested this on my iPhone 7 running iOS 14 and am unable to reproduce the issue.

**Comment by sylwester-** [comment]
iPhone 11 Pro (14.0.1) and iPad Air 2 (14.2 beta). I also have iPhone 6 with the older iOS version that I can try it on if you would like me to.

**Comment by liamdebeasi** [comment]
Let me see if any other team members can reproduce the issue. Will follow up when I have more info.

**Comment by sylwester-** [comment]
@liamdebeasi - I can confirm that the issue surfaces on iPhone 6 (iOS 12.4.8)

**Comment by sylwester-** [comment]
@liamdebeasi - can I help somehow with debugging? I listed devices above that the issue appears in.

**Comment by liamdebeasi** [comment]
Another team member was able to reproduce this issue last week. Are you able to see if Safari/WebKit is scrolling another element? An easy way to do this is to inspect the app in Safari Dev Tools, add a `scroll` event listener on the `window`, and log `event.target` in the callback. I am wondering if WebKit is latching onto some other element to scroll.

**Comment by liamdebeasi** [comment]
Apologies for the delay. I investigated this more and it appears to be a bug in iOS 14.

I reported the issue to the WebKit team here: https://bugs.webkit.org/show_bug.cgi?id=218533

I had reported a similar issue that affected iOS 12, so it makes sense that you are seeing this bug on iOS 12 also.

**Comment by sylwester-** [comment]
Thanks for this. There is a very ugly workaround that I'm using right now. 

On sidebar open I apply css style to ion-content and on sidebar close, I remove that style.

**Comment by liamdebeasi** [comment]
Another thing to try is to add a `z-index: 1` to `body` then remove it in a `rquestAnimationFrame`:

```js
requestAnimationFrame(() => {
  document.body.style.setProperty('z-index', '1');
  requestAnimationFrame(() => {
    document.body.style.removeProperty('z-index');
  });
});
```

You could do the above whenever the menu closes. I am in communication with the WebKit team now, so hopefully it gets resolved soon.

**Comment by ZakAttakk** [comment]
I have the same problem with the below app:
https://github.com/ZakAttakk/problemApp

Is anyone able to provided a more clear, detailed description of workarounds for this bug?  Perhaps taking into account my exact code?  I don't understand the above workarounds at all.

In my code, the "Search" and "Profile" pages do not work when visited via the side menu on iOS 14.3.

@liamdebeasi - For instance, more information about how to use "requestAnimationFrame" would be helpful.

@sylwester - What do you mean when you write, "On sidebar open I apply css style to ion-content and on sidebar close, I remove that style."

Thank you so much for you help.  This bug has been a thorn in my side for weeks.

Zach

**Comment by sylwester-** [comment]
@ZakAttakk - this is what I've done (very ugly):


export class AppComponent {
 
  fix_bug: string = '0px';

  ion_menu_scroll_bug() {
    this.fix_bug = '1px';
  }

  ion_menu_scroll_bug_fix() {
    this.fix_bug = '0px';
  }
}

<ion-menu side="start" contentId="main-content" menuId="side-menu" (ionDidClose)="ion_menu_scroll_bug_fix()" (ionDidOpen)="ion_menu_scroll_bug()">

...

<ion-menu>

<ion-router-outlet [style.top]="fix_bug" id="main-content"></ion-router-outlet>


Let me know if it helps. 

I tried different variations of this, including other ion-menu methods, however I found the above the most reliable. 

As you can see from the code, the html is in app-component.html

**Comment by ZakAttakk** [comment]
@sylwester- It worked!  Thank you so much.  I would never have figured that out on my own.  I converted what you did into React syntax:
```

ionMenuScrollBug = () => {
    this.setState({
      fixBug: "1px"
    })
  }

  ionMenuScrollBugFix = () => {
    this.setState({
      fixBug: "0px"
    })
  }

...

<IonRouterOutlet style={{top: this.state.fixBug}} id="main">

...

<IonMenu
          ref={this.sideMenuRef}
          contentId="main"
          max-edge-start="0"
          onIonDidClose={this.props.ionMenuScrollBugFix}
          onIonDidOpen={this.props.ionMenuScrollBug}
>

```

**Comment by sylwester-** [comment]
@ZakAttakk - glad it worked for you and thank you for converting this to React syntax. Hopefully it will help others!

Happy New Year!

**Comment by bmaciag** [comment]
Here is the solution for vue:
https://github.com/bmaciag/ionic-vue-webkit-scroll-issue/blob/master/src/App.vue

**Comment by liamdebeasi** [comment]
Hi everyone,

I can confirm that this issue is resolved in iOS 14.5 beta 1. I will follow up here when iOS 14.5 ships for all iOS users.

**Comment by liamdebeasi** [comment]
iOS 14.5 has shipped and includes a fix for this WebKit issue.

Thanks!
